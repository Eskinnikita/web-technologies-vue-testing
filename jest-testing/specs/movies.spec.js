import movies from "../models/movies"

describe('Movies', () => {
    let myMovies = []
    beforeEach(() => {
        myMovies = [{
            title: 'Age of Ultron',
            rate: null
        }]
    })

    test('can add a movie', () => {
        movies.add(myMovies, 'Kung Fury')
        expect(myMovies).toMatchSnapshot()
    })

    test('rate a movie', () => {
        movies.rate(myMovies[0], 5)
        expect(myMovies).toMatchSnapshot()
    })
})