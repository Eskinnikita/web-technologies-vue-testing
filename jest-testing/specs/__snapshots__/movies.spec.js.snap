// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies can add a movie 1`] = `
Array [
  Object {
    "rate": null,
    "title": "Age of Ultron",
  },
  Object {
    "rate": null,
    "title": "Kung Fury",
  },
]
`;

exports[`Movies rate a movie 1`] = `
Array [
  Object {
    "rate": 5,
    "title": "Age of Ultron",
  },
]
`;
